R/lockfile.R

```diff
@@ -212,7 +212,10 @@ renv_lockfile_create_impl <- function(project, type, libpaths, packages, exclude
   # warn if some required packages are missing
   ignored <- c(renv_project_ignored_packages(project), renv_packages_base(), exclude)
   missing <- setdiff(packages, c(names(records), ignored))
-  if (!the$status_running && !the$init_running)
+
+  # TODO: we likely need a better way to distinguish between top-level snapshot
+  # calls, versus snapshot calls that renv is using internally.
+  if (!the$status_running && !the$init_running && !the$restore_running)
     renv_snapshot_report_missing(missing, type)
```

Comment:
I'm curious about this, is this what causes snapshot to require input?

Reply:
That's right. But that prompt should only really be presented to the user when performing a top-level snapshot; that is, the user ran `renv::snapshot()` themselves. I think we need a cleaner way of handling this, but at least this is a good enough bandaid for now.

Reply:
Thank you so much for the explanation and your quick response on these issues! I really appreciate the work you do

Reply:
I'm happy I can help; sorry for the trouble in the interim!